portfolio/src/main/java/com/google/sps/data/CalendarGroup.java

```diff
@@ -0,0 +1,126 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+/** This class models a group of calendar events. */
+public class CalendarGroup {
+  public List<CalendarEvent> events;
+  public List<TimeRange> allFreeTimeRanges;
+
+  // These two variables represent the start and end of
+  // possible schedule time. When these times are passed in
+  // to instantiate a CalendarGroup class, these times are
+  // already adjusted based on user's working hour times
+  // and the current timestamp of when the user starts
+  // the scheduling process.
+  public final Instant overallStartTime;
+  public final Instant overallEndTime;
+
+  // Comparator for sorting events by start time
+  private static final Comparator<CalendarEvent> sortByEventStartTimeAscending =
+      Comparator.comparing(CalendarEvent::getStartTime);
+
+  /**
+   * @param events: a list of CalendarEvents' that represents the events already scheduled for the
+   *     user.
+   * @param startTime: start of possible scheduling blocks, of type Instant. This is the earlier of
+   *     working hour start time and the current timestamp.
+   * @param endTime: end of possible scheduling blocks, of type Instant.
+   */
+  public CalendarGroup(
+      List<CalendarEvent> events, Instant overallStartTime, Instant overallEndTime) {
+    if (events == null) {
+      throw new IllegalArgumentException("Calendar Group needs a collection of events");
```

Comment:
I think it might be simpler to make the exception message look like, "Events cannot be null". Here and below.

Reply:
Updated.